samples/dnn/siamrpnpp.py

```diff
@@ -0,0 +1,451 @@
+        im_patch = im_patch.transpose(2, 0, 1)
+        im_patch = im_patch[np.newaxis, :, :, :]
+        im_patch = im_patch.astype(np.float32)
+        
+        return im_patch
+    
+    def generate_anchor(self, score_size):
+        anchors = Anchors(self.anchor_stride, self.anchor_ratios, self.anchor_scales)
+        anchor = anchors.anchors
+        x1, y1, x2, y2 = anchor[:, 0], anchor[:, 1], anchor[:, 2], anchor[:, 3]
+        anchor = np.stack([(x1 + x2) * 0.5, (y1 + y2) * 0.5, x2 - x1, y2 - y1], 1)
+        total_stride = anchors.stride
+        anchor_num = anchor.shape[0]
+        anchor = np.tile(anchor, score_size * score_size).reshape((-1, 4))
+        ori = - (score_size // 2) * total_stride
+        xx, yy = np.meshgrid([ori + total_stride * dx for dx in range(score_size)],
+                             [ori + total_stride * dy for dy in range(score_size)])
+        xx, yy = np.tile(xx.flatten(), (anchor_num, 1)).flatten(), \
+                 np.tile(yy.flatten(), (anchor_num, 1)).flatten()
+        anchor[:, 0], anchor[:, 1] = xx.astype(np.float32), yy.astype(np.float32)
+        return anchor
+
+    def _soft_max(self, x):
+        x_max = x.max(0)
+        e_x = np.exp(x - x_max)
+        y = e_x / e_x.sum(axis=0)
+        return y
+    
+    def _convert_bbox(self, delta, anchor):
+        delta_transpose = np.transpose(delta, (1, 2, 3, 0))
+        delta_contig = np.ascontiguousarray(delta_transpose)
+        delta_view = delta_contig.reshape(4, -1)
+        delta = delta_view
+
+        delta[0, :] = delta[0, :] * anchor[:, 2] + anchor[:, 0]
+        delta[1, :] = delta[1, :] * anchor[:, 3] + anchor[:, 1]
+        delta[2, :] = np.exp(delta[2, :]) * anchor[:, 2]
+        delta[3, :] = np.exp(delta[3, :]) * anchor[:, 3]
+
+        return delta
+    
+    def _convert_score(self, score):
+        score_transpose = np.transpose(score, (1, 2, 3, 0))
+        score_contig = np.ascontiguousarray(score_transpose)
+        score_view = score_contig.reshape(2, -1)
+        score = np.transpose(score_view, (1, 0))
+        score = self._soft_max(score)
+        score = score[:,1]
+        return score
+    
+    def _bbox_clip(self, cx, cy, width, height, boundary):
+        cx = max(0, min(cx, boundary[1]))
+        cy = max(0, min(cy, boundary[0]))
+        width = max(10, min(width, boundary[1]))
+        height = max(10, min(height, boundary[0]))
+        return cx, cy, width, height
+    
+    def tracker_template(self, z):
+        backbone_target = self.backbone_target
+        backbone_target.setInput(z)
+        cv_resnet_target_output_1 = backbone_target.forward('output')
+        cv_resnet_target_output_2 = backbone_target.forward('458')
+        cv_resnet_target_output_3 = backbone_target.forward('490')
+
+        neck_1_out_1 = self.neck_1_out_1
+        neck_1_out_1.setInput(cv_resnet_target_output_1)
+        cv_neck_1_target_output = neck_1_out_1.forward('output')
```

Comment:
When you need output from the last layer, you can use forward without args.
```suggestion
       self.zf_1 = neck_1_out_1.forward()
```